Use typed methods such as setInt, setDouble, setDate ... for prepared statement

tpeyrard opened the pull request:
Use typed methods such as `setInt`, `setDouble`, `setDate` ... for prepared statement.

It moves parsing from the driver to Jmeter, but it allows to compare more easily two drivers / databases. 
Moreover it makes benchmark closer to reality, in fact for performance reasons we try to avoid calls to setObject as the driver as to cast/parse String parameters.

tpeyrard committed:
Use typed methods such as setInt, setDouble, setDate ... for prepared statements instead of calling every time setObject.

 It moves parsing from the driver to Jmeter, but it allows to compare more easily two drivers / databases.
 Moreover it makes benchmark closer to reality, in fact for performance reasons we try to avoid calls to setObject as the driver as to cast/parse String parameters.

milamberspace commented on src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java:
```diff
@@ -46,6 +28,14 @@
+import java.sql.Date;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
```
Please use explicit import, don't use wildcard import.

vlsi commented on src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java:
```diff
@@ -338,6 +328,44 @@ private static final int countRows(ResultSet resultSet) throws SQLException {
+            break;
+        case Types.DOUBLE:
+        case Types.DECIMAL:
+            pstmt.setDouble(index, Double.parseDouble(argument));
```
This is not valid.

>https://docs.oracle.com/javase/1.5.0/docs/guide/jdbc/getstart/mapping.html
>The recommended Java mapping for the DECIMAL and NUMERIC types is java.math.BigDecimal

Commit:
- Expand imports java.sql.*
- Expand imports java.util.*
- Fix conversion from JDBC Types to Java Types according to Oracle documentation (Number, BigDecimal and Float)
- Add more conversion types

Samoht-fr commented:
I fixed patch according to your remarks. I also added new conversions according to the [Oracle doc] (https://docs.oracle.com/javase/1.5.0/docs/guide/jdbc/getstart/mapping.html).

vlsi commented on src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java:
```diff
@@ -18,34 +18,44 @@
+import org.apache.jmeter.threads.JMeterVariables;
+import org.apache.jmeter.util.JMeterUtils;
+import org.apache.jorphan.logging.LoggingManager;
+import org.apache.log.Logger;
```
Please refrain from moving imports around. Can you please move `org.apache` imports back to the initial location?

Samoht-fr replied:
That's my IntelliJ ... I'll fix it :)

vlsi commented:
Looks good except from a glitch with imports moved around.
I'm not sure if additional tests are expected in this PR.

Commit:
- Remove unused imports

milamberspace commented:
Thanks. In the trunk.

URL: http://svn.apache.org/r1700060
Log:
Use typed methods such as setInt, setDouble, setDate ... for prepared statement #27

Bugzilla Id: 58301

This closes #27


Modified:
    jmeter/trunk/src/protocol/jdbc/org/apache/jmeter/protocol/jdbc/AbstractJDBCTestElement.java
    jmeter/trunk/xdocs/changes.xml

Samoht-fr commented:
Thanks too.
